### DISCUSSION TITLE
I need a bit of help

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#348] [Q&A] [answered]
I'm trying to help on https://github.com/orbitalquark/textadept/issues/344 and https://github.com/orbitalquark/textadept/issues/347. I don't know why he closed the issues but I'm still working on them. He was right about the rules for modula_hex and modula_real. They are not working. I'm trying to fix but I have no idea how to: https://github.com/iqskpduswupkcjqg/lite-xl-modula-lexers/blob/main/modula2.lua

Update: I fixed it, simply move modula_hex and modula_real before modula_octal and modula_dec. I also fixed Modula-3 numeric literals: https://github.com/iqskpduswupkcjqg/lite-xl-modula-lexers/blob/main/modula3.lua

**Comment by orbitalquark** [comment]
Please provide example code that is failing to highlight properly along 
with the expected output. I know absolutely nothing about those 
languages and don't know what `modula_hex` and `modula_real` are 
supposed to match or not match.

**Reply by ghost** [reply]
I fixed them all @orbitalquark, the Modula lexers are almost done. What I'm currently stuck is how to correctly highlight string interpolation on Objeck. This is how Objeck's string interpolation looks like: https://objeck.org/getting_started.html#strings and this is my Objeck lexer: https://github.com/iqskpduswupkcjqg/textadept-objeck-lexer/blob/main/objeck.lua. Please help.

**Comment by orbitalquark** [comment, marked as answer]
Unfortunately, there is no way to easily add string interpolation. It is 
a difficult problem to solve without embedded lexers. I'm sorry that I 
cannot be of help here.

**Reply by ghost** [reply]
OK. So my Objeck lexer is also ready: https://github.com/iqskpduswupkcjqg/textadept-objeck-lexer/blob/main/objeck.lua

**Comment by orbitalquark** [comment]
Okay, thanks. This is considered a "legacy" lexer and will have to be 
converted before being committed: 
https://orbitalquark.github.io/scintillua/api.html#migrating-legacy-lexers

I can do this myself when I have some time. Please let me know the file 
extension(s) for Objeck for auto-detection.

**Reply by ghost** [reply]
Objeck uses .obs extension.

**Comment by paaguti** [comment]
hi folks

I'm offline for personal reasons until the end of February. If you forked
from my github for the Modula-x lexers, send me PR's (I will really
appreciate them) and I'll integrate in March. No cycles to look at the hex
numbers now...

TIA, /PA

On Sat, 28 Jan 2023 at 20:49, iqskpduswupkcjqg ***@***.***>
wrote:

> Objeck uses .obs extension.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/348#discussioncomment-4805658>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4DVE3I6SXHP2KJJVBLWUVZ3XANCNFSM6AAAAAAUJT4BOU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Comment by orbitalquark** [comment]
Added in Scintillua: https://github.com/orbitalquark/scintillua/commit/00dfea4b8cc3ce4e371e009c7bb17da50c4d8a27

Thank you!
